pkg/utils/url.go

```diff
@@ -51,3 +51,22 @@ func JoinURL(baseURL, relativeURL string) (string, error) {
 	// Return the joined URL as a string
 	return parsedBaseURL.String(), nil
 }
+
+// ContainsRegistry returns true if the specified registryHost is part of registries
+func ContainsRegistry(registries []string, registryHost string) bool {
+	cleanRegistryURL := func(u string) string {
+		u = strings.TrimSpace(u)
+		u = strings.TrimPrefix(u, "http://")
+		u = strings.TrimPrefix(u, "https://")
+		return strings.Split(u, "/")[0]
+	}
+	registryHost = cleanRegistryURL(registryHost)
+
+	for _, reg := range registries {
+		reg = cleanRegistryURL(reg)
+		if strings.HasSuffix(reg, registryHost) {
```

Comment:
Good question: I am also thinking on what should be the right thing to do here. Let me try to explain the thinking and see if there is any better way we can find.

The user might have a registry that requires authentication like `packages.repo.com`. But images are fetched from multiple different sub URLs like `project1.packages.repo.com`, `projects2.packages.repo.com` etc. To avoid users specifying all the registry hosts, I was thinking with this check we can allow users to only specify `packages.repo.com` and it will apply to all the above-mentioned hosts.

Reply:
I think to achieve what you describe the arguments to `HasSuffix()` might need to be reversed?

That being said, looking at the new production vs staging uri, they seem to have the same suffix but one of them needs authentication while the other does not.  So I was wondering if we should use the suffix idea or not. I guess using it gives more flexibility and in our particular case we could put the full URI for the repo that needs authentication. I’m not really sure though. What do you think?

Reply:
After giving a bit more thought, I am also inclining towards switching back to `strings.EqualFold` which will require users to provide full registry hostname. I think it would be better to be explicit instead of assuming how the underlying registry does authentication.